core/src/main/java/io/lionweb/lioncore/java/serialization/NodePopulator.java

```diff
@@ -98,19 +128,28 @@ private void populateNodeReferences(
                         Node referred =
                             (Node) classifierInstanceResolver.resolve(entry.getReference());
                         if (entry.getReference() != null && referred == null) {
-                          switch (serialization.getUnavailableReferenceTargetPolicy()) {
-                            case NULL_REFERENCES:
-                              referred = null;
-                              break;
-                            case PROXY_NODES:
-                              referred = deserializationStatus.resolve(entry.getReference());
-                              break;
-                            case THROW_ERROR:
-                              throw new DeserializationException(
-                                  "Unable to resolve reference to "
-                                      + entry.getReference()
-                                      + " for feature "
-                                      + serializedReferenceValue.getMetaPointer());
+
+                          // For LionCore Builtins, we want to automatically update references,
+                          // using Resolve Info
+                          Node autoresolvedElement = autoResolveMap.get(entry.getResolveInfo());
```

Comment:
Why only if `entry.reference != null`? We said for a multi-LW-version compatible language, we'd serialize with only the resolveInfo, no reference.

Reply:
You are right, we should do this auto-resolve also when entry.reference is null. I changed that.

Regarding your second point, my understanding was that we had the option of looking at the reference field, and, if and only if that is not set or if it points to a node that cannot be found, we auto-rewire it to the element of the builtins that is available in the current version. So I do not expect `referred` to be necessarily null and in general I would populate it, as it would make the reference behaves "normally" in most situations, with the special behavior being relevant only during migrations.

Reply:
I'm unsure about the "right" behavior here. Probably needs some experimentation.

Reply:
I added three tests for NodePopulator to show that this would work when:
- We refer to a builtin element with its expected id for this version
- We refer to a builtin element with the id it had in another version (i.e., actually with any "wrong" id)
- We refer to a builtin element without specifying the id but just the name